src/bin/spin.rs

```diff
@@ -38,26 +38,26 @@ fn version() -> &'static str {
 
 /// The Spin CLI
 #[derive(Parser)]
-#[clap(
-    name = "spin",
+#[command(
+    id = "spin",
     version = version(),
 )]
 enum SpinApp {
-    #[clap(subcommand, alias = "template")]
+    #[command(subcommand, alias = "template")]
```

Comment:
(Not a comment on this particular line.)

Using this opportunity to point out to other reviewers that clap 4.0.0+ has updated default display order.  Whereas in 3.x.x the commands are alphabetically sorted, 4+ appears to defer to the order we declare commands here in code.

### Previous to this PR (looking at `SUBCOMMANDS`):

<img width="566" alt="Screenshot 2023-01-30 at 4 43 25 PM" src="https://user-images.githubusercontent.com/9322017/215622031-974353ec-1439-42b5-b992-c9a951d4aafb.png">

### With this PR (looking at `Commands`):

<img width="537" alt="Screenshot 2023-01-30 at 4 43 06 PM" src="https://user-images.githubusercontent.com/9322017/215622151-69135c8a-5692-47e3-803d-323a3aed3375.png">

@itowlson (and others) -- Do we have a preference on ordering?  If so, should we ensure commands are listed here in the order we'd like them displayed?  (Or, I imagine there is a clap 4+ setting we might utilize, regardless of order in code.)

Reply:
Eek, good catch.  I'd be inclined to go for alphabetical - some tools (such as Cargo do try to group similar commands together, but I'm not sure if we have a strong enough notion of that at the moment.  And if we do it, we should approach it intentionally rather than as part of a package rev.

Reply:
Should I do anything here? Change the order by hand, force the old way, leave it be for now?

Using a flag would be fighting clap quite a bit as it wants to be WYSIWYG going forward.

```
New behavior:

    Sorted order is derived by default
    Sorting is turned on by cmd.next_display_order(None)
        This is not recursive, it must be set on each level
    The display order incrementing is mixed with arguments
        This does make it slightly more difficult to predict
```

from: https://github.com/clap-rs/clap/pull/3975

Reply:
Maybe the best bet is to change the order by hand, assuming we'd like to retain the current alphabetical order.